src/clr-addons/readonly/readonly.directive.ts

```diff
@@ -0,0 +1,112 @@
+import { AfterViewChecked, Directive, ElementRef, Injector, Input, Renderer2 } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+@Directive({
+  selector: '[clrReadonly]',
+})
+export class ClrReadonlyDirective implements AfterViewChecked {
+  @Input('clrMulti') isMultiSelect: boolean = false;
+  @Input('clrUnitPosition') unitPosition: 'left' | 'right' = 'right';
+  @Input('clrReadOnlyProperty') arrayPosition: string | null = null;
+  @Input('clrReadonly') clrReadOnly: boolean = true;
+
+  private readonly originalElement: HTMLElement;
+  private isInitialized = false;
+  private isList: boolean = false;
+  private isNumeric: boolean = false;
+  private ngControl: NgControl;
+
+  constructor(
+    private elementRef: ElementRef,
+    private readonly renderer: Renderer2,
+    private readonly injector: Injector
+  ) {
+    this.originalElement = elementRef.nativeElement;
```

Comment:
either use originalElement everywhere instead of elementRef.nativeElement or remove originalElement at all